framework/src/executioners/PicardSolve.C

```diff
@@ -99,6 +99,9 @@ PicardSolve::validParams()
   params.addParam<bool>("update_xfem_at_timestep_begin",
                         false,
                         "Should XFEM update the mesh at the beginning of the timestep");
+  params.addParam<bool>("auto_advance",
+                        "Whether to automatically advance sub-applications regardless of whether "
+                        "their solve converges.");
```

Comment:
It might be a good idea to document what use case requires us to advance the state even though sub-apps fail to solve?

Reply:
I don't know what those use cases are. As you know I hate multiapps :smile:

Reply:
Do you, @YaqiWang, or @vincentlaboure know of some? I agree that I should add documentation about those cases. Otherwise yea even I don't understand why it's there!

Reply:
I can't think of an example where `auto_advance=true` would be desired but it doesn't mean there isn't one

Reply:
This does sound counter logic. If the solve in suapp is not successful, why not stop but rather advance?

Reply:
why not name the parameter as `force_advance` and default it to `false`?

Reply:
The current logic auto advances sub-applications as long as you are not doing Picard. If I do what you are describing, then I will be changing the default behavior. Presumably we have tests and/or applications that rely on this default behavior. I don't know how this dumpster fire advanced to the point where we are at now, but I am terrified of modifying default behavior. I am shocked any time I make changes in the PicardSolve/Transient/TransientMultiApp code system, and I don't break something.

Reply:
You could have more shock before I created PicardSolve ;-) This is exactly we want to have tests. Breaking tests will force us to think of the design.

Reply:
The test I added applies `auto_advance = false`. All other test cases in the world test the other case.

Reply:
OK, the conclusion is that we still do not know why we have that option when subapps fail